packages/generators/app/src/index.ts

```diff
@@ -80,17 +149,27 @@ export const generateNewApp = (projectDirectory: string, options: Partial<NewOpt
     });
   });
 
-  parseDatabaseArguments({ scope, args: options });
   initCancelCatcher();
 
-  return generateNew(scope).catch((error) => {
-    console.error(error);
+  try {
+    await trackUsage({ event: 'willCreateProject', scope });
+
+    const scopeExtension = await parseDatabaseArguments(options);
+    merge(scope, scopeExtension);
+
+    // create a project with full list of questions
+    return await createProject(scope);
+  } catch (error) {
+    if (!(error instanceof Error)) {
+      throw error;
+    }
```

Comment:
why not tracking errors that are not of Error type?

Reply:
ts is complaining if the type isn't and instance of error, it's just a safety measure we should always receive errors anyway but still need a runtime check to make ts happy